### ISSUE TITLE
Promisify .validate model method

### ISSUE THREAD

**Issue by nmagerko** [balderdashy/waterline#1089]
In my Sails controllers, I want to validate requests to create a new model instance against my Waterline models themselves. To do so is rather trivial; all I need to do is call 

```
Model.validate(data, function(err, data){
    // omitted
})`, 
```

where `Model` is a Waterline model.

However, it seems somewhat strange to me that a `validate` class/static method that returns a Bluebird promise (as the CRUD methods do) is not available. Instead, I find myself having to add `var validateAsync = Promise.promisify(Model.validate);` (where `Promise` is the Bluebird promise library) and then calling `validateAsync(data).then(...` all over the place.

Would it be possible to include an async validation method so that I don't have to keep doing this? Nodebacks are painful, especially when I have multiple models to validate, one after another. Of course, if I completely missed something in the documentation, I apologize. 

<hr />

Additionally, I'd really love to see custom error messages on my model validations. If there's anything that I can do to help get that done, please let me know.

**Comment by particlebanana** [comment]
@nmagerko if you would like to help with the bluebird integration that would be great! 

The two places to look are the [Validate Method](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/validate.js) and then the validate logic will need to be added to [Deferred Object](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/deferred.js). You can see an example on the [Find One](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/basic.js#L48-L51) function.

**Comment by nmagerko** [comment]
Awesome! I'll start taking a look.

**Comment by sailsbot** [comment]
Thanks for posting, @nmagerko.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
